metals/src/main/scala/scala/meta/internal/metals/debug/DebugProxy.scala

```diff
@@ -95,6 +107,32 @@ private[debug] final class DebugProxy(
         .map(DebugProtocol.syntheticResponse(request, _))
         .foreach(client.consume)
 
+    case request @ CompletionRequest(args) =>
+      val completions = for {
+        frame <- lastFrames.find(_.getId() == args.getFrameId())
+        originalSource = frame.getSource()
+        sourceUri = clientAdapter.toMetalsPath(originalSource.getPath)
```

Comment:
You could move assignments to `yield` block because after desugaring they translate into additional `map { case (...) }`.

Reply:
Changed